src/services/teams/hooks/helpers.js

```diff
@@ -0,0 +1,481 @@
+        else
+            return a1.reduce((stack, element) => {
+                const v1 = element[key];
+                for (let i = 0; i < a2.length; i++) {
+                    if (v1 === a2[i][key])
+                        return stack;            //if found return without adding
+                }
+                stack.push(element);              //if not found add element from a1
+                return stack;
+            }, []);
+    };
+
+    return diff(oldArray, newArray);
+};
+
+/**
+ * @helper
+ * @requires function:arrayDiff
+ * @param {Array::*} baseArray 
+ * @param {Array::*} changedArray 
+ * @param {String} [key] - optional for objects in arrays
+ * @return {Object::{remove:[],add:[]} }
+ */
+const arrayRemoveAddDiffs = (baseArray, changedArray, key) => {
+    return { remove: arrayDiff(baseArray, changedArray, key), add: arrayDiff(changedArray, baseArray, key) };
+};
+
+/**
+ * @helper 
+ * @private
+ * @requires function::createUserWithRole
+ * @param {Object::hook} hook
+ * @param {Array::(stringId||Object::TeamUser||Object::String)} teamUsers 
+ * @param {Object::Team}
+ */
+const mappedInputUserIdsToTeamUsers = (hook, teamUsers, oldTeam) => {
+    if (!isArray(teamUsers))
+        throw new errors.BadRequest('param teamUsers must be an array', teamUsers);
+
+    const teamownerRoleId = hook.findRole('name', 'teamowner', '_id');
+    const teamowner = oldTeam.userIds.find(_user => isSameId(_user.role, teamownerRoleId));
+
+    if (isUndefined(teamowner))
+        throw new errors.NotAcceptable('No teamowner found for this team.');
+
+    return teamUsers.map((e) => {
+        let role, userId;
+        //userId
+        if (hasKey(e, 'userId'))
+            userId = bsonIdToString(e.userId);
+
+        if (isObjectId(e) || isString(e))
+            userId = bsonIdToString(e);
+
+        //teamowner role can not repatched at the moment todo: later test if any other has this role, after map 
+        if (isSameId(teamowner.userId, userId))
+            return { userId, role: teamownerRoleId };
+
+        //roleId
+        if (hasKey(e, 'role')) {
+            if (isObjectIdWithTryToCast(e.role))
+                role = e.role;
+            else
+                role = hook.findRole('name', e.role, '_id');
+        } else
+            role = hook.findRole('name', 'teammember', '_id');
+
+        role = bsonIdToString(role);
+
+
+
+        if (isUndefined(userId) || isUndefined(role))
+            throw new errors.BadRequest('Can not mapped userIds to teamUsers', { userId, role });
+
+        return { userId, role };
+    });
+};
+
+/**
+ * @helper
+ * @private
+ * @param {Array::teamUser} teamUsers 
+ * @param {Array||String} userIds 
+ * @return {Array::teamUser}
+ */
+const removeTeamUsers = (teamUsers, userIds) => {
+    if (isString(userIds))
+        userIds = [userIds];
+
+    if (!isArrayWithElement(userIds))
+        return teamUsers;
+
+    return teamUsers.reduce((list, teamUser) => {
```

Comment:
Is this not just a simple filter operation? `teamUser.filter(user => !userIds.includes(user.userId))`

Reply:
done